src/taipy/core/common/typing.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from typing import TypeVar, Union
+
+ModelType = TypeVar("ModelType")
+Entity = TypeVar("Entity")
```

Comment:
there's a class called `_Entity` in file `src/taipy/core/common/_entity.py` that all entities extended, I think it will come to be handy for this :D

Reply:
Yes, it was intended to remove this functions elsewhere. We can see where as we go alone. The idea is to decouple the base class to maintain only the responsability of communicating with the database itself.